node/src/components/block_validator/tests.rs

```diff
@@ -0,0 +1,282 @@
+            panic!("unexpected event: {:?}", reactor_event);
+        }
+    }
+
+    async fn expect_fetch_deploy<T>(&self, deploy: T)
+    where
+        T: Into<Option<Deploy>>,
+    {
+        let (reactor_event, _) = self.scheduler.pop().await;
+        if let ReactorEvent::Fetcher(FetcherRequest::Fetch {
+            id,
+            peer,
+            responder,
+        }) = reactor_event
+        {
+            match deploy.into() {
+                None => responder.respond(None).await,
+                Some(deploy) => {
+                    assert_eq!(id, *deploy.id());
+                    let response = FetchResult::FromPeer(Box::new(deploy), peer);
+                    responder.respond(Some(response)).await;
+                }
+            }
+        } else {
+            panic!("unexpected event: {:?}", reactor_event);
+        }
+    }
+}
+
+fn new_proposed_block(
+    timestamp: Timestamp,
+    deploy_hashes: Vec<DeployHash>,
+    transfer_hashes: Vec<DeployHash>,
+) -> ProposedBlock<ClContext> {
+    // Accusations and ancestors are empty, and the random bit is always true:
+    // These values are not checked by the block validator.
+    let block_context = BlockContext::new(timestamp, vec![]);
+    let block_payload = BlockPayload::new(deploy_hashes, transfer_hashes, vec![], true);
+    ProposedBlock::new(block_payload, block_context)
+}
+
+fn new_deploy(rng: &mut TestRng, timestamp: Timestamp, ttl: TimeDiff) -> Deploy {
+    let secret_key = SecretKey::random(rng);
+    let chain_name = "chain".to_string();
+    let payment = ExecutableDeployItem::ModuleBytes {
+        module_bytes: Bytes::new(),
+        args: runtime_args! { ARG_AMOUNT => U512::from(1) },
+    };
+    let session = ExecutableDeployItem::ModuleBytes {
+        module_bytes: Bytes::new(),
+        args: RuntimeArgs::new(),
+    };
+    let dependencies = vec![];
+    let gas_price = 1;
+
+    Deploy::new(
+        timestamp,
+        ttl,
+        gas_price,
+        dependencies,
+        chain_name,
+        payment,
+        session,
+        &secret_key,
+    )
+}
+
+fn new_transfer(rng: &mut TestRng, timestamp: Timestamp, ttl: TimeDiff) -> Deploy {
+    let secret_key = SecretKey::random(rng);
+    let chain_name = "chain".to_string();
+    let payment = ExecutableDeployItem::ModuleBytes {
+        module_bytes: Bytes::new(),
+        args: runtime_args! { ARG_AMOUNT => 1 },
+    };
+    let session = ExecutableDeployItem::Transfer {
+        args: RuntimeArgs::new(),
+    };
+    let dependencies = vec![];
+    let gas_price = 1;
+
+    Deploy::new(
+        timestamp,
+        ttl,
+        gas_price,
+        dependencies,
+        chain_name,
+        payment,
+        session,
+        &secret_key,
+    )
+}
+
+/// Validates a block using a `BlockValidator` component, and returns the result.
+async fn validate_block(
+    rng: &mut TestRng,
+    timestamp: Timestamp,
+    deploys: Vec<Deploy>,
+    transfers: Vec<Deploy>,
+) -> bool {
```

Comment:
Note: this will not model the case in which a block in invalid because we could not fetch its deploys.

New JIRA ticket: https://casperlabs.atlassian.net/browse/NDRS-1153